main.go

```diff
@@ -0,0 +1,357 @@
+		Visibility      string `yaml:"visibility"`
+	} `yaml:"createOpts"`
+}
+
+// NodeImages represents the structure of the config.yaml file.
+type NodeImages struct {
+	APIVersion          string               `yaml:"apiVersion"`
+	OpenStackNodeImages []OpenStackNodeImage `yaml:"openStackNodeImages"`
+}
+
+const (
+	provider        = "openstack"
+	outputDirectory = "./output"
+)
+
+func usage() {
+	fmt.Printf(`%s create-node-images cluster-stack-directory cluster-stack-release-directory [node-image-registry-path]
+This command is a csctl plugin.
+https://github.com/SovereignCloudStack/csctl
+`, os.Args[0])
+}
+
+func main() {
+	minArgs := 4
+	maxArgs := 5
+	if len(os.Args) < minArgs || len(os.Args) > maxArgs {
+		fmt.Printf("Wrong number of arguments. Expected %d or %d, got %d\n", minArgs, maxArgs, len(os.Args))
+		usage()
+		os.Exit(1)
+	}
+	if os.Args[1] != "create-node-images" {
+		usage()
+		os.Exit(1)
+	}
+	clusterStackPath := os.Args[2]
+	csctlConfig, err := csctlclusterstack.GetCsctlConfig(clusterStackPath)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	configFilePath := filepath.Join(clusterStackPath, "node-images", "config.yaml")
+	config, err := GetConfig(configFilePath)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	if csctlConfig.Config.Provider.Type != provider {
+		fmt.Printf("Wrong provider in %s. Expected %s\n", clusterStackPath, provider)
+		os.Exit(1)
+	}
+	releaseDir := os.Args[3]
+	_, err = os.Stat(releaseDir)
+	if err != nil {
+		fmt.Println(err.Error())
+		os.Exit(1)
+	}
+	registryConfigPath := os.Args[4]
+
+	method := csctlConfig.Config.Provider.Config["method"]
+	switch method {
+	case "get":
+		// Copy config.yaml to releaseDir as node-images.yaml
+		dest := filepath.Join(releaseDir, "node-images.yaml")
+		if err := copyFile(configFilePath, dest); err != nil {
+			fmt.Printf("Error copying config.yaml to releaseDir: %v\n", err)
+			os.Exit(1)
+		}
+		fmt.Println("config.yaml copied to releaseDir as node-images.yaml successfully!")
+	case "build":
+		for _, image := range config.OpenStackNodeImages {
+			if image.ImageDir == "" {
+				fmt.Printf("No images to build, image directory is not defined in config.yaml file")
+				os.Exit(1)
+			}
+
+			// Construct the path to the image folder
+			packerImagePath := filepath.Join(clusterStackPath, "node-images", image.ImageDir)
+
+			if _, err := os.Stat(packerImagePath); err != nil {
+				fmt.Printf("Image folder %s does not exist\n", packerImagePath)
+				os.Exit(1)
+			}
+			fmt.Println("Running packer build...")
+			// Warning: variables like build_name and output_directory must exist in packer ariables file like in example
+			// #nosec G204
+			cmd := exec.Command("packer", "build", "-var", "build_name="+image.ImageDir, "-var", "output_directory="+outputDirectory, packerImagePath)
```

Comment:
you could think about running this in parallel

Reply:
Good idea, we can do that also for the `pushToS3` function and the whole process from building to pushing images to S3 should take less time to finish.

Reply:
you can call it in separate goroutines. That will not be a problem at all!

Reply:
Sure. But what can happen is that we currently support packer builders(plugins), which run locally like `qemu`. If there are multiple images and we run them in parallel, we can hit the hardware limits of the cluster-stack creator and it will fail, right? Maybe it's just an edge case and it won't happen in almost any use case.

Reply:
Did you get some new insights on this? I guess that I would always prefer running this on proper hardware rather than waiting n times longer. But this certainly depends on the use case. @jschoone I think you might also have an opinion

Reply:
Nothing new from my side, but I agree that parallel execution is better when you have the right hardware. I would suggest doing it in another PR after this one is merged. But I would also wait for Jan's opinion.

Reply:
sounds good to me!

Reply:
@jschoone what do you think about this approach to build images in parallel?

Reply:
Hi @janiskemper and @michal-gubricky,
generally I'm fine with parallel image building. I think it's hard to consider the hardware because it can even be to few for one build.
But: do we need to implement this on our own? AFAIK packer handles it itself when multiple builds are defined in the `image.json.pkr`.
Or the question is more: what happens if you pass that kind of build instruction, is the csctl-plugin able to upload the multiple outputs of packer?

Reply:
> Hi @janiskemper and @michal-gubricky, generally I'm fine with parallel image building. I think it's hard to consider the hardware because it can even be to few for one build. But: do we need to implement this on our own? AFAIK packer handles it itself when multiple builds are defined in the `image.json.pkr`. Or the question is more: what happens if you pass that kind of build instruction, is the csctl-plugin able to upload the multiple outputs of packer?

No, csctl-plugin currently cannot push multiple outputs of the packer, when you define it in one `image.json.pkr`. It is expected that `one image=one image folder`

Reply:
Based on this discussion, the [issue#13](https://github.com/SovereignCloudStack/csctl-plugin-openstack/issues/13) was created.